ql/src/java/org/apache/hadoop/hive/ql/parse/ImportSemanticAnalyzer.java

```diff
@@ -818,6 +803,20 @@ private static void createReplImportTasks(
         throw new SemanticException(ErrorMsg.DATABASE_NOT_EXISTS.getMsg(tblDesc.getDatabaseName()));
       }
     }
+
+    if (table != null) {
+      if (!replicationSpec.allowReplacementInto(table)) {
+        // If the target table exists and is newer or same as current update based on repl.last.id, then just noop it.
+        return;
+      }
+    } else {
+      // If table doesn't exist, allow creating a new one only if the database state is older than the update.
+      if ((parentDb != null) && (!replicationSpec.allowReplacementInto(parentDb))) {
```

Comment:
my suggestion was replicationSpec.isOlderThan() as in the comment above, not sure if i missed something there,